keps/sig-release/20190316-rebase-images-to-distroless.md

```diff
@@ -0,0 +1,149 @@
+---
+title: Rebase Kubernetes Main Master and Node Images to Distroless
+authors:
+  - "@yuwenma"
+owning-sig: sig-auth
+participating-sigs:
+  - sig-auth
+reviewers:
+  - "@tallclair"
+approvers:
+  - "@tallclair"
+editor: yuwenma
+creation-date: 2019-03-16
+last-updated: 2019-03-16
+status: implementable
+---
+
+# Rebase Kubernetes Main Master and Node Images to Distroless
+
+## Table of Contents
+
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Background -- How the goal and scope are determined based on reality](#background)
+    *  [Kubernetes-Images](#kubernetes-images)
+        *  [Type 1 From Scratch](#type-1-from-scratch)
+        *  [Type 2 Debian Based Images](#type-2-debian-based-images)
+        *  [Type 3 Alpine Based Images](#type-3-alpine-based-images)
+    *  [Distroless and Previous Work](#distroless-and-previous-work)
+* [Proposal](#proposal)
+* [Graduation Criteria](#graduation-criteria)
+* [Testing](#testing)
+    * [Test the Core Images](#test-the-core-images)
+    * [Test the Addon Containers](#test-the-addon-containers)
+* [Implementation History](#implementation-history)
+
+## Summary
+
+The effort of rebasing the k8s images to distroless is aimed at making the k8s images thinner, safer and less vulnerable. The scope is not only improving the core containers but will cover the master and node addons which have their own release process. As for the core containers, this effort is targeting the v1.15 release.
+## Motivation
+
+Rebasing the k8s images to distroless can make the images thinner, safer and less vulnerable.
+
+Meanwhile, it will drastically reduce churn on the total number of k8s images versions. Due to the fact that many images are based on debian base and a vulnerability in debian base (a couple times a month) will result in rebuilding every image, changing the image from debian base to distroless static can reduce the total number of k8s images.
+
+What's more, it reduces the burden of managing and maintaining multiple k8s images from the security (e.g. CVE), compatibility and build process concerns.
+### Goals
+
+Use image gcr.io/distroless/static:latest as the only base image for the following K8s/GKE images
+
+- Images based FROM scratch
+- Images based on debian/alpine and only for the purpose of redirecting logs with shell.
+- Images based on debian due to previous rebasing from busybox.
+
+Document the full list of distroless-based images as well as the exception list
```

Comment:
It might be good to define the scope a little more precisely. How do we decide which images these changes apply to?

One goal we might set out is to have a policy that says, for an image to be hosted in the official k8s.gcr.io image repository, it must be based on either distroless/scratch or debian-base. And, if it is based on debian base it must be documented as an exception with a justification.

Reply:
Good suggestion! Thanks!